lib/services/auth-service.js

```diff
@@ -0,0 +1,149 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var di = require('di');
+var jwt = require('jsonwebtoken');
+var crypto = require('crypto');
+
+module.exports = authServiceFactory;
+di.annotate(authServiceFactory, new di.Provide('Auth.Services'));
+di.annotate(authServiceFactory,
+    new di.Inject(
+        'Assert',
+        'Services.Configuration'
+    )
+);
+
+function authServiceFactory(
```

Comment:
@cgx027 I think we should look at https://github.com/ncb000gt/node.bcrypt.js for password hashing and verification.